File: Python Labs/com/pattern/flyweigh/ClothingItemFactory.py
from com.pattern.flyweigh.SharedClothingItem import SharedClothingItem


class ClothingItemFactory:
    clothing_item_map = {}

    @staticmethod
    def get_shared_clothing_item(item_type):
        if item_type not in ClothingItemFactory.clothing_item_map:
            ClothingItemFactory.clothing_item_map[item_type] = SharedClothingItem(item_type)
        return ClothingItemFactory.clothing_item_map[item_type]

File: Java Labs/Patterns On Java/src/com/pattern/flyweigh/ClothingItemFactory.java
package com.pattern.flyweigh;

import java.util.HashMap;
import java.util.Map;

public class ClothingItemFactory {
    private static final Map<String, ClothingItem> clothingItemMap = new HashMap<>();

    public static ClothingItem getSharedClothingItem(String type) {
        return clothingItemMap.computeIfAbsent(type, SharedClothingItem::new);
    }
}
